flang/test/Examples/hello.f90

```diff
@@ -3,7 +3,7 @@
 ! default when they are available.
 
 ! RUN: bbc %s -o - | tco | llc --relocation-model=pic --filetype=obj -o %t.o
-! RUN: %CC %t.o -L%L -lFortranRuntime -lFortranDecimal -lm -o hello
+! RUN: %CC %t.o -L%L -lFortranRuntime -lFortranDecimal -lFortran_main -lm -o hello
```

Comment:
I can reproduce the issue. I find it strange that the order matters but I can fix it.

Reply:
I think it's because a library A that require symbol x from a library B must be linked before linking B, so that the linker knows it needs symbol x from B when linking B (https://stackoverflow.com/questions/45135/why-does-the-order-in-which-libraries-are-linked-sometimes-cause-errors-in-gcc).

Here, Fortran_main is the only library that uses `_FortranAProgramStart`. So if FortranRuntime is linked before, the linker won't include `_FortranAProgramStart` because it does not yet know it needs it and this causes a failure in the end.

I guess these constraints on the user allow the linker to make a single pass and be faster.